designs/0031-deprecation-schedule.md

```diff
@@ -0,0 +1,254 @@
+deprecated, but have been left for major version change to actually remove. This
+Stan 3.0 version seems to consistently be just over the horizon, and as a result
+some features, like the above-mentioned `<-` operator, have been deprecated for
+5+ years. Having the ability to remove or correct these tiny features allows the
+growth of Stan as a language to continue without an ever-increasing amount of
+difficulty.
+
+This is not a theoretical problem. Consider [this
+issue](https://github.com/stan-dev/stanc3/issues/953) with the parser or [this
+discussion](https://discourse.mc-stan.org/t/list-of-stanc3-new-reserved-keywords/12948)
+of new keywords.
+
+These are issues that stem from either the inability to make minor changes to
+the language, or from the current lack of process to introduce these changes.
+The solution to both of them is rather simple from a user perspective - the new
+`array[]` syntax can be automatically produced by our pretty-printer, and it is
+easy to rename a variable, but this must be clearly and consistently
+communicated to our users.
+
+Finally, this process provides an opportunity for feedback on changes. The
+deprecation period prior to removal is not only to let users know of our
+intentions, but also to let them tell us if our intentions need to be rethought.
+By communicating the changes and putting deadlines on them, we create a sort of
+community review period during which we can recieve feedback and course-correct
+if we have made a poor choice.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+Deprecated minor features may be removed from Stan during minor version changes.
+Here, 'minor change' is restricted to be changes which can be done automatically
+through the use of the stanc3 canonicalizer, or a user-driven find and replace
+operation. By their very nature, minor changes are primarily syntactic. This is
+in contrast to 'major changes', which are by their nature semantic, and require
+human-in-the-loop reasoning to accommodate. Major changes, such the interface
+changes to the differential equation solvers, may be announced via deprecation
+warnings in minor versions, but will not be implemented/removed prior to a
+version 3.0.0.
+
+A minor feature must be deprecated for a minimum of 3 minor versions
+(approximately 1 year on the current release schedule) before removal. The exact
+timeline for a feature's removal will be decided during the development which
+would mark it as deprecated.
+
+A minor feature is deprecated and a candidate for future removal when:
+1. stanc emits a compile-time warning with the following 3 parts:
+    1. A description of the deprecated feature, such as "Use of <- for assignment
+       is deprecated".
+    2. An _action_ the user should take, such as "Use = instead". If stanc3 can
+       automate this for them, we should call that out, leading to a full
+       message like "Use = instead or run stanc3 --print-canonical to update
```

Comment:
I think the stanc3 command needs a path to the file name.  And it should mention what's being replaced.  It's also not clear what the boundary of the command is, so I'd put it last, like this:

```
Use = instead of <- for assignment;  this can be automatically fixed by
running: stanc3 --print-canonical <stan-program-file>
```